libcloud/common/linode.py

```diff
@@ -163,3 +176,119 @@ def add_default_params(self, params):
         # Be explicit about this in case the default changes.
         params["api_responseFormat"] = "json"
         return params
+
+
+class LinodeExceptionV4(Exception):
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return "%s" % self.message
+
+    def __repr__(self):
+        return "<LinodeException '%s'>" % self.message
+
+
+class LinodeResponseV4(JsonResponse):
+    valid_response_codes = [httplib.OK, httplib.NO_CONTENT, ]
+
+    def parse_body(self):
+        """Parse the body of the response into JSON objects
+        :return: ``dict`` of objects"""
+        return super(LinodeResponseV4, self).parse_body()
+
+    def parse_error(self):
+        """
+        Parse the error body and raise the appropriate exception
+        """
+        status = int(self.status)
+        data = self.parse_body()
+        # Use only the first error, as there'll be only one most of the time
+        error = data['errors'][0]
```

Comment:
Why not:
error = " ".join(data['errors'])
In this case all error messages will be returned, not only the first one.

Reply:
data['errors'] is a list of dictionaries so I think this will not work.
Maybe it could be something like this?
```suggestion
        error_list = []
        for error in data['errors']:
            reason = error.get('reason')
            # The field in the request that caused this error
            field = error.get('field')
            if field is not None:
                msg = '%s-%s' % (reason, field)
            else:
                msg = reason
            error_list.append(msg)
        error_msg = ' '.join(error_list)
```

Reply:
Yes, you are right. May be you can leave the original one. As you said in the first error is the most important one.